OutOfSchool/OutOfSchool.DataAccess/Repository/WorkshopRepository.cs

```diff
@@ -45,5 +46,23 @@ public new async Task Delete(Workshop entity)
 
         /// <inheritdoc/>
         public bool ClassExists(long id) => db.Classes.Any(x => x.Id == id);
+
+        /// <inheritdoc/>
+        public async Task<IEnumerable<Workshop>> GetListOfWorkshopsForSynchronizationByOperation(ElasticsearchSyncOperation operation)
+        {
+            var elasticsearchSyncRecords = db.ElasticsearchSyncRecords;
+
+            var resultMaxDates = from record in elasticsearchSyncRecords
+                                 group record by record.RecordId into groupedRecords
+                                 select new { RecordId = groupedRecords.Key, MaxOperationDate = groupedRecords.Max(r => r.OperationDate) };
+
+            var result = from rMaxDate in resultMaxDates
+                         join record in elasticsearchSyncRecords on new { rMaxDate.RecordId, OperationDate = rMaxDate.MaxOperationDate } equals new { record.RecordId, record.OperationDate } into leftJoin
+                         from joinedRecord in leftJoin
+                         where joinedRecord.Operation == operation
+                         select new { joinedRecord.RecordId, joinedRecord.OperationDate, joinedRecord.Operation };
+
+            return await db.Workshops.Join(result, workshop => workshop.Id, result => result.RecordId, (workshop, result) => workshop).ToListAsync();
```

Comment:
1. From join I see that `RecordId` is FK field to primary key `WorkshopId`. 
2. I don't see this in migration script.

So, either naming is wrong (and `RecordId` should be named as `WorkshopId` and db schema should be fixed) or extra filtering condition is missed (that has to specify that only workshops' data are taken from `ElasticsearchSyncRecords`)

Similar logic is present in `WorkshopServicesCombiner` - where workshop id is stored as record id without any additional type info

Reply:
yes, at first I added additional field "TableName", I consider that in feature there are possible different entities, but for now I decided to simplify table. not sure that that was a correct decision. I will fix it.